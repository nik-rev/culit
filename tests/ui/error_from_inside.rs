//! Each literal emits a type mismatch error.
//! I expect the span of this error to be exactly at the literal

use culit::culit;

struct Kilometer(u32);

#[culit]
fn main() {
    0w;
    0.0w;
    'x'w;
    b'x'w;
    "x"w;
    b"x"w;
    c"x"w;
}

mod custom_literal {
    macro_rules! w {
        ($value:literal $($tt:tt)*) => {{
            let it: () = $value;
            it
        }};
    }
    pub(crate) use w;

    pub mod int {
        pub(crate) use super::w;
    }

    pub mod float {
        pub(crate) use super::w;
    }

    pub mod str {
        pub(crate) use super::w;
    }

    pub mod char {
        pub(crate) use super::w;
    }

    pub mod byte_char {
        pub(crate) use super::w;
    }

    pub mod byte_str {
        pub(crate) use super::w;
    }

    pub mod c_str {
        pub(crate) use super::w;
    }
}
