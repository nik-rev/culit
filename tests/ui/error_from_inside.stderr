error[E0308]: mismatched types
  --> tests/ui/error_from_inside.rs:10:5
   |
10 |     0w;
   |     ^^ expected `()`, found `&str`
...
22 |             let it: () = $value;
   |                     -- expected due to this

error[E0308]: mismatched types
  --> tests/ui/error_from_inside.rs:11:5
   |
11 |     0.0w;
   |     ^^^^ expected `()`, found `&str`
...
22 |             let it: () = $value;
   |                     -- expected due to this

error[E0308]: mismatched types
  --> tests/ui/error_from_inside.rs:12:5
   |
12 |     'x'w;
   |     ^^^^ expected `()`, found `char`
...
22 |             let it: () = $value;
   |                     -- expected due to this

error[E0308]: mismatched types
  --> tests/ui/error_from_inside.rs:13:5
   |
13 |     b'x'w;
   |     ^^^^^ expected `()`, found integer
...
22 |             let it: () = $value;
   |                     -- expected due to this

error[E0308]: mismatched types
  --> tests/ui/error_from_inside.rs:14:5
   |
14 |     "x"w;
   |     ^^^^ expected `()`, found `&str`
...
22 |             let it: () = $value;
   |                     -- expected due to this

error[E0308]: mismatched types
  --> tests/ui/error_from_inside.rs:15:5
   |
15 |     b"x"w;
   |     ^^^^^ expected `()`, found `&[u8; 1]`
...
22 |             let it: () = $value;
   |                     -- expected due to this

error[E0308]: mismatched types
  --> tests/ui/error_from_inside.rs:16:5
   |
16 |     c"x"w;
   |     ^^^^^ expected `()`, found `&CStr`
...
22 |             let it: () = $value;
   |                     -- expected due to this
